src/datadog/hex.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+// This component provides a function, `hex`, for formatting an integral value
+// in hexadecimal.
+
+#include <cassert>
+#include <charconv>
+#include <limits>
+#include <system_error>
+#include <utility>
+
+namespace datadog {
+namespace tracing {
+
+// Return the specified `value` formatted as a lower-case hexadecimal string
+// without any leading zeroes.
+template <typename Integer>
+std::string hex(Integer value) {
+  // 4 bits per hex digit char, and then +1 char for possible minus sign
```

Comment:
Negative hex values shouldn't be a thing here, since it's intended to be an encoding of a bit pattern (more like a hexdump) instead of a representation of a numeric value.